src/app/api/v1/endpoints/extract_data.py

```diff
@@ -0,0 +1,290 @@
+            )
+
+        # Convert the bounding box to PNG coordinates and return the response
+        return ExtractCoordinatesResponse(
+            bbox=extracted_coords.bbox.rescale(
+                original_height=pdf_page_height,
+                original_width=pdf_page_width,
+                target_height=png_page_height,
+                target_width=png_page_width,
+            ),
+            coordinates=Coordinates(
+                east=extracted_coords.coordinates.east,
+                north=extracted_coords.coordinates.north,
+                projection=extracted_coords.coordinates.projection,
+            ),
+        )
+    elif extract_data_request.format == FormatTypes.ELEVATION:
+        extracted_elevation: ExtractElevationResponse | None = extract_elevation(
+            extract_data_request, pdf_page, user_defined_bbox.to_fitz_rect()
+        )
+
+        if extracted_elevation is None:
+            return NotFoundResponse(
+                detail="Elevation not found.",
+                bbox=extract_data_request.bbox,
+            )
+
+        # Convert the bounding box to PNG coordinates and return the response
+        return ExtractElevationResponse(
+            bbox=extracted_elevation.bbox.rescale(
+                original_height=pdf_page_height,
+                original_width=pdf_page_width,
+                target_height=png_page_height,
+                target_width=png_page_width,
+            ),
+            elevation=extracted_elevation.elevation,
+        )
+
+    elif extract_data_request.format == FormatTypes.TEXT:
+        extracted_text: ExtractTextResponse = extract_text(pdf_page, user_defined_bbox.to_fitz_rect())
+
+        # Convert the bounding box to PNG coordinates and return the response
+        return ExtractTextResponse(
+            bbox=extracted_text.bbox.rescale(
+                original_height=pdf_page_height,
+                original_width=pdf_page_width,
+                target_height=png_page_height,
+                target_width=png_page_width,
+            ),
+            text=extracted_text.text,
+        )
+    elif extract_data_request.format == FormatTypes.NUMBER:
+        extracted_number = extract_number(pdf_page, user_defined_bbox.to_fitz_rect())
+
+        # Convert the bounding box to PNG coordinates and return the response
+        return ExtractNumberResponse(
+            bbox=extracted_number.bbox.rescale(
+                original_height=pdf_page_height,
+                original_width=pdf_page_width,
+                target_height=png_page_height,
+                target_width=png_page_width,
+            ),
+            number=extracted_number.number,
+        )
+    else:
+        raise ValueError("Invalid format type.")
+
+
+def extract_coordinates(
+    extract_data_request: ExtractDataRequest, pdf_page: fitz.Page, user_defined_bbox: fitz.Rect
+) -> ExtractDataResponse | None:
+    """Extract coordinates from a PNG image.
```

Comment:
At this point, we are not really concerned with the PNG image anymore, but we are rather extracting from the PDF page, I would say. So maybe we should change the docstring accordingly, here and for the other methods as well.

Reply:
I tried to clarify the docstring in general. I think it is easy to get lost between the user-defined bbox in PNG coordinates and the PDF coordinates and what is give to what function.